packages/suite/src/actions/wallet/send/sendFormThunks.ts

```diff
@@ -80,44 +90,144 @@ export const importSendFormRequestThunk = createThunk(
 );
 
+const updateRbfLabelsThunk = createThunk(
+    `${MODULE_PREFIX}/updateReplacedTransactionThunk`,
+    (
+        {
+            labelsToBeEdited,
+            precomposedTx,
+            txid,
+        }: {
+            labelsToBeEdited: RbfLabelsToBeUpdated;
+            precomposedTx: PrecomposedTransactionFinal;
+            txid: string;
+        },
+        { dispatch, extra },
+    ) => {
+        const {
+            thunks: { moveLabelsForRbfAction },
+        } = extra;
+
+        if (labelsToBeEdited !== undefined) {
+            dispatch(
+                moveLabelsForRbfAction({
+                    toBeMovedOrDeletedList: labelsToBeEdited,
+                    newTxid: txid,
+                }),
+            );
+        }
+
+        // notification from the backend may be delayed.
+        // modify affected transaction(s) in the reducer until the real account update occurs.
+        // this will update transaction details (like time, fee etc.)
+        dispatch(
+            replaceTransactionThunk({
+                precomposedTx,
+                newTxid: txid,
+            }),
+        );
+    },
+);
+
+const applySendFormMetadataLabelsThunk = createThunk(
+    `${MODULE_PREFIX}/applyMetadataLabelsThunk`,
+    (
+        {
+            selectedAccount,
+            precomposedTx,
+            txid,
+        }: {
+            selectedAccount: Account;
+            precomposedTx: GeneralPrecomposedTransactionFinal;
+            txid: string;
+        },
+        { dispatch, getState },
+    ) => {
+        const metadata = selectMetadata(getState());
+        if (metadata.enabled) {
+            const precomposedForm = selectPrecomposedSendForm(getState());
+            let outputsPermutation: number[];
+            if (isCardanoTx(selectedAccount, precomposedTx)) {
+                // cardano preserves order of outputs
+                outputsPermutation = precomposedTx?.outputs.map((_o, i) => i);
+            } else {
+                outputsPermutation = precomposedTx?.outputsPermutation;
+            }
+
+            const synchronize = getSynchronize();
+
+            precomposedForm?.outputs
+                // create array of metadata objects
+                .map((formOutput, index) => {
+                    const { label } = formOutput;
+                    // final ordering of outputs differs from order in send form
+                    // outputsPermutation contains mapping from @trezor/utxo-lib outputs to send form outputs
+                    // mapping goes like this: Array<@trezor/utxo-lib index : send form index>
+                    const outputIndex = outputsPermutation.findIndex(p => p === index);
+                    const outputMetadata: Extract<MetadataAddPayload, { type: 'outputLabel' }> = {
+                        type: 'outputLabel',
+                        entityKey: selectedAccount.key,
+                        txid, // txid becomes available, use it
```

Comment:
what does this comment mean? txid should always be there, shouldnt it?

Reply:
fixed in e3583e3c18